### ISSUE TITLE
Receiving Push Notification After App Data is Cleared

### ISSUE THREAD

**Issue by winsonloh1998** [OneSignal/OneSignal-Flutter-SDK#590]
**Description:**

I'm not really sure is this the intended/ correct behavior, but it shouldn't be getting the notification after the data is cleared in my opinion. 

If it's really work in this way, is there any solution/ workaround to prevent it from receiving the notification once the user cleared the data? 

**Environment**

OneSignal Flutter Version : 3.4.0
Flutter Version: 3.0.5
Added SDK by following this link: https://documentation.onesignal.com/docs/android-sdk-setup

**Steps to Reproduce Issue:**

1. Go to the app settings and pressed clear data.
2. Push a notification to that android device.
3. Receive the notification. 

**Anything else:**

So far i have tested it only in Android as this is my project requirement.
I have tried this in emulator as well as physical device (Samsung S9 - Android 10)

**Comment by nan-li** [comment]
Hi @winsonloh1998,

Thank you for reporting this behavior. I was able to reproduce this too: clear app data, and receive a notification (after clearing data but before opening the app). I agree this feels incorrect, but it appears to be the existing FCM behavior. 

I found an [old OneSignal issue](https://github.com/OneSignal/react-native-onesignal/issues/967) that discusses this, as well as [an issue reported in Firebase's flutterfire](https://github.com/firebase/flutterfire/issues/3016) with no resolution.

I searched online about this behavior and looked for documentation, but was not able to find any helpful information. Thus, we don't have any workaround or solution as this seems to be the expected behavior.

It is worth noting that on Android 13 devices, clearing the app data does clear any notification permissions the app has, and no notifications will be displayed as a result.
